src/View/ViewContainer.vala

```diff
@@ -296,36 +302,55 @@ namespace Marlin.View {
             refresh_slot_info (slot.location);
         }
 
-        public void user_path_change_request (GLib.File loc, bool allow_mode_change = true, bool make_root = true) {
+        private void user_path_change_request (GLib.File loc, bool allow_mode_change = true, bool make_root = true) {
             /* Ony call directly if it is known that a change of folder is required
              * otherwise call focus_location.
              */
-            view.user_path_change_request (loc, allow_mode_change, make_root);
+            open_location (loc,
+                           make_root ? Marlin.OpenFlag.NEW_ROOT : Marlin.OpenFlag.DEFAULT,
+                           allow_mode_change);
         }
 
-        private void on_slot_new_container_request (GLib.File loc, Marlin.OpenFlag flag = Marlin.OpenFlag.NEW_ROOT) {
+        private void open_location (GLib.File loc,
+                                    Marlin.OpenFlag flag = Marlin.OpenFlag.NEW_ROOT,
+                                    bool allow_mode_change = true) {
+
+            bool change_to_icon_mode = false;
```

Comment:
I'm not seeing this change anywhere else removed, can we get an explanation why, it's only present here in this diff?